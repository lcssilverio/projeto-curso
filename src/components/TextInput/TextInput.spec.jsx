import { render, screen } from "@testing-library/react";
import { TextInput } from ".";
import userEvent from "@testing-library/user-event";


describe("Tests to TextInput component", () => {
  test("should have a value of searchValue", () => {
    const fn = jest.fn()
    render(<TextInput handleChange={fn} searchValue={'testando'}/>);

    const input = screen.getByPlaceholderText(/type your search/i)
    expect(input.value).toBe('testando')
  });

  test("should call handleChange function on each key pressd", () => {
    const fn = jest.fn()
    render(<TextInput handleChange={fn}/>);

    const input = screen.getByPlaceholderText(/type your search/i)

    const value = 'o valor'

    userEvent.type(input, value)

    expect(input.value).toBe(value)
    expect(fn).toHaveBeenCalledTimes(value.length)
  });

  test("should match snapshot", () => {
    const fn = jest.fn()
    const {container} = render(<TextInput handleChange={fn} searchValue={'testando'}/>);
    expect(container).toMatchSnapshot()
  });
});
