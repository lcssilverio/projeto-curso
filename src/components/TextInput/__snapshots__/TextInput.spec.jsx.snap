// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to Posts component should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
